src/jdk.compiler/share/classes/com/sun/tools/javac/comp/ArgumentAttr.java

```diff
@@ -718,6 +729,36 @@ Type methodType() {
         }
     }
 
+    /**
+     * Argument type for default values.
+     */
+    class ResolvedDefaultType extends ResolvedMemberType<JCDefaultValue> {
+
```

Comment:
I am surprised that ResolvedDefaultType extends ResolvedMemberType rather than ArgumentType directly. While the Inference behavior could be indirectly modelled it as if the class has a generic factory method named 'default':
```
<T> static Foo<T> default() { ... }
```
this implementation detail could be pushed deep down and not manifest here and in  elsewhere (setPolyKind())